albyhub/docker-compose.yml

```
@@ -0,0 +1,22 @@
+version: "3.7"
+services:
+  app_proxy:
+    environment:
+      APP_HOST: albyhub_server_1
+      APP_PORT: 8080
+      PROXY_AUTH_ADD: "false"
+  server:
+    image: ghcr.io/getalby/hub:v1.6.0@sha256:45ec367860a8f42c6f66317dd5889d8e2af8b0df1e0bc59b33e67c08770e4e17
+    volumes:
```

If this container can run as a non-root user (which should be possible based on this Dockerfile https://github.com/getAlby/hub/blob/master/Dockerfile) then we can change this to:

```suggestion
  server:
    image: ghcr.io/getalby/hub:v1.6.0@sha256:45ec367860a8f42c6f66317dd5889d8e2af8b0df1e0bc59b33e67c08770e4e17
    user: "1000:1000"
    volumes:
```

When the app is installed, umbrelOS creates the data directory that's commited to this repo with 1000:1000 ownership/permissions, so a `getalby/hub` container running as user 1000 will have write permissions to `${APP_DATA_DIR}/data`

---

My docker and docker compose knowledge is limited. We do not want to change the image now as it is used in various places and I would be worried to break other deployments. 

The app only writes to the configured volume. (log files and SQLite DB files) 
If we configure this user here what does that exactly mean? can you point me to some docs? 

It also binds to ports > 1024 - 8080 for the webapp and the lightning port 9375 (if the embedded LDK node is used and not the default Umbrel LND node) - so no root is needed for that.

---

No worries. The `user` key in a docker-compose.yml file allows you to run the container as a certain user WITHOUT having to change the underlying image.

Setting `user` in the docker-compose.yml overrides the user used to run the container process. If you don't set a `user` in the docker-compose.yml file then Docker uses whatever has been defined in the Dockerfile with the `USER` instruction. In the case of your [Dockerfile](https://github.com/getAlby/hub/blob/master/Dockerfile), the lack of `USER` instruction means that by default it runs as root.

As long as the container can write to it's log and db files as user 1000:1000 then we should be good. I can give this a test for you.

Docs:
https://docs.docker.com/reference/dockerfile/#user
https://docs.docker.com/reference/compose-file/services/#user